utils/auth/scopes.ts

```diff
@@ -1,44 +1,276 @@
-export const Scopes = {
-  USER_READ_EMAIL: BigInt(1) << BigInt(0),
-  USER_READ: BigInt(1) << BigInt(1),
-  USER_WRITE: BigInt(1) << BigInt(2),
-  USER_DELETE: BigInt(1) << BigInt(3),
-  USER_AUTH_WRITE: BigInt(1) << BigInt(4),
-  NOTIFICATION_READ: BigInt(1) << BigInt(5),
-  NOTIFICATION_WRITE: BigInt(1) << BigInt(6),
-  PAYOUTS_READ: BigInt(1) << BigInt(7),
-  PAYOUTS_WRITE: BigInt(1) << BigInt(8),
-  ANALYTICS: BigInt(1) << BigInt(9),
-  PROJECT_CREATE: BigInt(1) << BigInt(10),
-  PROJECT_READ: BigInt(1) << BigInt(11),
-  PROJECT_WRITE: BigInt(1) << BigInt(12),
-  PROJECT_DELETE: BigInt(1) << BigInt(13),
-  VERSION_CREATE: BigInt(1) << BigInt(14),
-  VERSION_READ: BigInt(1) << BigInt(15),
-  VERSION_WRITE: BigInt(1) << BigInt(16),
-  VERSION_DELETE: BigInt(1) << BigInt(17),
-  REPORT_CREATE: BigInt(1) << BigInt(18),
-  REPORT_READ: BigInt(1) << BigInt(19),
-  REPORT_WRITE: BigInt(1) << BigInt(20),
-  REPORT_DELETE: BigInt(1) << BigInt(21),
-  THREAD_READ: BigInt(1) << BigInt(22),
-  THREAD_WRITE: BigInt(1) << BigInt(23),
-  PAT_CREATE: BigInt(1) << BigInt(24),
-  PAT_READ: BigInt(1) << BigInt(25),
-  PAT_WRITE: BigInt(1) << BigInt(26),
-  PAT_DELETE: BigInt(1) << BigInt(27),
-  SESSION_READ: BigInt(1) << BigInt(28),
-  SESSION_DELETE: BigInt(1) << BigInt(29),
-  PERFORM_ANALYTICS: BigInt(1) << BigInt(30),
-  COLLECTION_CREATE: BigInt(1) << BigInt(31),
-  COLLECTION_READ: BigInt(1) << BigInt(32),
-  COLLECTION_WRITE: BigInt(1) << BigInt(33),
-  COLLECTION_DELETE: BigInt(1) << BigInt(34),
-  ORGANIZATION_CREATE: BigInt(1) << BigInt(35),
-  ORGANIZATION_READ: BigInt(1) << BigInt(36),
-  ORGANIZATION_WRITE: BigInt(1) << BigInt(37),
-  ORGANIZATION_DELETE: BigInt(1) << BigInt(38),
-  SESSION_ACCESS: BigInt(1) << BigInt(39),
+const scopeDefinitions = [
+  {
+    name: 'USER_READ_EMAIL',
+    description: 'Read your email',
```

Comment:
Would be great to see localization support for these since they'll probably also need a slight refactor, but overall looks good